astropy/table/table.py

```diff
@@ -842,8 +842,8 @@ def __getitem__(self, item):
             return self.columns[item]
         elif isinstance(item, (int, np.integer)):
             return self.Row(self, item)
-        elif isinstance(item, (tuple, list)) and all(isinstance(x, six.string_types)
-                                                     for x in item):
+        elif (isinstance(item, (tuple, list)) and item and
```

Comment:
I had a feeling you would comment on this and nearly wrote a pre-emptive response.  :smile:  As it was I decided that for this update it would be correct to stay within the bounds of the current API which doesn't allow arbitrary iterable classes.  If that seems desirable we can address that separately.